File: Solutions/872. Leaf-Similar Trees/872.java
class Solution {
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        ArrayList<Integer> al1 = new ArrayList<>();
        ArrayList<Integer> al2 = new ArrayList<>();
        getLeaves(root1, al1);
        getLeaves(root2, al2);
        if(al1.size() != al2.size()){
            return false;
        }
        for(int i = 0; i < al1.size(); i++){
            if(al1.get(i) != al2.get(i)){
                return false;
            }
        }
        return true;
    }
    public void getLeaves(TreeNode root, ArrayList<Integer> al){
        if(root == null){
            return;
        }
        if(root.left == null && root.right == null){
            al.add(root.val);
        } 
        getLeaves(root.left, al);
        getLeaves(root.right, al);
    }
}

File: Solutions/872. Leaf-Similar Trees/872.py
class Solution:
    def leafSimilar(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> bool:
        arr1 = []
        arr2 = []
        self.getLeaf(root1, arr1)
        self.getLeaf(root2, arr2)
        if len(arr1) != len(arr2):
            return False
        for i in range(len(arr1)):
            if arr1[i] != arr2[i]:
                return False
        return True

    def getLeaf(self, root: Optional[TreeNode], arr: List[int]):
        if root is None:
            return
        if root.left is None and root.right is None:
            arr.append(root.val)
        self.getLeaf(root.left, arr)
        self.getLeaf(root.right, arr)
        